### PR TITLE
labels: add missing labels

### LINKED ISSUE #149: created VM is missing important labels and an annotation
These are the labels and one annotation which are required by the UI:

VM created from example template in UI:

```yaml
apiVersion: kubevirt.io/v1alpha3
kind: VirtualMachine
metadata:
  annotations:
    kubevirt.io/latest-observed-api-version: v1alpha3
    kubevirt.io/storage-observed-api-version: v1alpha3
    name.os.template.kubevirt.io/fedora31: Fedora 31
  name: asd
  namespace: default
  labels:
    app: asd
    flavor.template.kubevirt.io/tiny: 'true'
    os.template.kubevirt.io/fedora31: 'true'
    vm.kubevirt.io/template: vm-template-example
    vm.kubevirt.io/template.namespace: default
    vm.kubevirt.io/template.revision: '1'
    vm.kubevirt.io/template.version: v0.9.1
    workload.template.kubevirt.io/server: 'true'
spec:
  running: false
  template:
    metadata:
      creationTimestamp: null
      labels:
        flavor.template.kubevirt.io/tiny: 'true'
        kubevirt.io/domain: asd
        kubevirt.io/size: tiny
        os.template.kubevirt.io/fedora31: 'true'
        vm.kubevirt.io/name: asd
        workload.template.kubevirt.io/server: 'true'
    spec:
      domain:
        cpu:
          cores: 1
          sockets: 1
          threads: 1
        devices:
          disks:
            - bootOrder: 1
              disk:
                bus: virtio
              name: containerdisk
            - disk:
                bus: virtio
              name: cloudinitdisk
          interfaces:
            - macAddress: '02:dd:94:00:00:02'
              masquerade: {}
              model: virtio
              name: default
          networkInterfaceMultiqueue: true
          rng: {}
        machine:
          type: q35
        resources:
          requests:
            memory: 1Gi
      hostname: asd
      networks:
        - name: default
          pod: {}
      terminationGracePeriodSeconds: 0
      volumes:
        - containerDisk:
            image: 'kubevirt/fedora-cloud-container-disk-demo:latest'
          name: containerdisk
        - cloudInitNoCloud:
            userData: |-
              #cloud-config
              password: fedora
              chpasswd: { expire: False }
          name: cloudinitdisk

```

### PR DESCRIPTION
**Pull request by pkliczewski**
Fixes https://github.com/kubevirt/vm-import-operator/issues/149

Signed-off-by: Piotr Kliczewski <piotr.kliczewski@gmail.com>

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -126,6 +126,13 @@ func MakeLabelFrom(name string) string {
 	return name
 }
 
+// AddLabels add labels
```
Comment (jakub-dzon) [inline pkg/utils/utils.go]: "adds"

File: pkg/templates/template-handler.go [hunk 2]
```diff
@@ -63,4 +66,5 @@ func addLabels(vm *kubevirtv1.VirtualMachine, template *templatev1.Template) {
 		vm.ObjectMeta.SetLabels(labels)
 	}
 	labels[templateNameLabel] = template.GetObjectMeta().GetName()
+	labels[templateNamespace] = template.GetObjectMeta().GetNamespace()
```
Comment (masayag) [inline pkg/templates/template-handler.go]: This caused an issue to validate the VM on creation after vmSpec was created by tempalte-validator.
Please check if the entire process can be completed on an a cluster that runs OCP + common templates that can be resolved.

See commit #158 3fa098ed8fb4758f44029fec2579831c51a3eb7c

File: pkg/providers/ovirt/provider.go [hunk 3]
```diff
@@ -225,7 +226,27 @@ func (o *OvirtProvider) FindTemplate() (*templatev1.Template, error) {
...
+	if err != nil {
+		return nil, err
+	}
+	return updateLabels(vm, labels, *vmName), nil
+}
+
+func updateLabels(vm *kubevirtv1.VirtualMachine, labels map[string]string, vmName string) *kubevirtv1.VirtualMachine {
+	utils.AddLabels(vm.ObjectMeta.GetLabels(), labels)
+
+	labels[vmNameLabel] = vmName
```
Comment (masayag) [inline pkg/providers/ovirt/provider.go]: this seems common to all providers - can we move it to https://github.com/kubevirt/vm-import-operator/pull/158/files#diff-c79b0981c3f458c6eaedf56f00a4e063R69 (pkg/templates/template-handler.go#addLabels )

File: pkg/templates/template-handler.go (hunk 2 above)
Comment (pkliczewski) [reply pkg/templates/template-handler.go]: OK, I will remove

File: pkg/providers/ovirt/provider.go (hunk 3 above)
Comment (pkliczewski) [reply pkg/providers/ovirt/provider.go]: Update and the label list is common but the way how we obtain the information to build the label map is specific. I thought about it and failed to find good way to do it.

**Review by jakub-dzon — APPROVED** [review]

File: pkg/providers/ovirt/provider.go (hunk 3 above)
Comment (masayag) [reply pkg/providers/ovirt/provider.go]: I only referred to "vm.kubevirt.io/name" label.
There is a set of labels that we create in pkg/templates/template-handler.go#addLabels
"vm.kubevirt.io/name" requires the value of vmName which is already passed to o.templateHandler.ProcessTemplate(template, vmName, namespace) in line 229.

Comment (pkliczewski) [reply pkg/providers/ovirt/provider.go]: OK, will move "vm.kubevirt.io/name" label.

**Review by masayag — APPROVED** [review]

**Pull request merged**